plugins/inputs/jolokia/jolokia.go

```diff
@@ -252,7 +252,14 @@ func (j *Jolokia) Gather(acc telegraf.Accumulator) error {
 							switch t2 := v.(type) {
 							case map[string]interface{}:
 								for k2, v2 := range t2 {
-									fields[measurement+"_"+k+"_"+k2] = v2
+									switch t3 := v2.(type) {
+									case map[string]interface{}:
+										for k3, v3 := range t3 {
+											fields[measurement+"_"+k+"_"+k2+"_"+k3] = v3
+										}
```

Comment:
Holy nesting batman!

I think a programmatic solution would be better, not an explicit 3-key map.

For example:
```
func extractValues(k string, v interface{}, fields map[string]interface{}) {
	if v, ok := v.(map[string]interface{}); ok {
		for k2, v2 := range v {
			extractValues(k+"_"+k2, v2, fields)
		}
	} else {
		fields[k] = v
	}
}

func Gather() {
	...
	if values, ok := out["value"]; ok {
		extractValues(measurement, values, fields)
	}
	...
}
```